Lexionic: add theme

henriqueiamarino opened the pull request:
Lexiconic is a sleek, versatile theme that empowers book authors, publishers, and storytellers. Its particular layout makes it easy to create stunning landing pages and offers a seamless way to showcase a piece of writing.

[Demo site](https://lexionic.mystagingwebsite.com/) | [Site content export](https://github.com/user-attachments/files/18574513/lexionic.wordpress.2025-01-10.xml_.zip)


![screenshot](https://github.com/user-attachments/assets/eff8333d-54fd-4e86-9477-af2752e8b135)

henriqueiamarino committed:
Lexionic: add theme

iamtakashi commented:
@henriqueiamarino Here are my notes. The number of things I need to mention is getting less and less. Great job!

* I've noticed that the custom spacing sizes defined in the `theme.json` aren't working. If the JSON file is version 3, the `defaultSpacingSizes` must be `false` to override the default spacing size. [Example](https://github.com/WordPress/twentytwentyfive/blob/trunk/theme.json#L58). I reviewed the theme as if the custom spacing sizes are actually working, but please fix it in the next update.
* Most of the custom duotones aren't used in the theme, do we need them?
* I like the unique-looking CTA in the footer, but it's not looking great on small screens. Can we take a different approach to make it work on both large and small screen sizes?
![CleanShot 2025-01-29 at 14 42 11@2x](https://github.com/user-attachments/assets/1124e4a9-033d-4263-b9e8-2cfbfd32a529)
* It's not a problem, but why do the images on the blog home template use the cover blocks but not the image blocks?
* It's probably better to use a heading for the text "Homer". It's awkward in the variations that use a font with a strong character for headings. 
![CleanShot 2025-01-29 at 15 17 44@2x](https://github.com/user-attachments/assets/337e4da0-f610-4fdb-8d36-69185e3b4034)
* A `main` tag is missing in all templates.
* Remove the following unused patterns. `patterns/single.php`, `patterns/page.php`, 
* All the style variations have custom duotones. Do we actually need them? If we do, it's probably better if they were registered in the default json.
* Change the `Requires PHP:` to `7.2` in both `readme.txt` and `style.css`
* Let's add credits to the rest of two images bundled with the theme.
* Thanks for paying attention to the theme tags! You can also add `block-styles` and `full-site-editing`.

henriqueiamarino committed:
Lexionic: fixes following review

henriqueiamarino commented:
Thanks @iamtakashi. Although the check has failed, I believe I fixed all the issues.
Here's the checklist:

- I tried to fix the custom spacing sizes issue by adding the line `defaultSpacingSizes: false`. I am not sure if this is enough. Please let me know;
- I removed all the duotones. They belong to an initial idea that is now gone. I did it on the style variations, as well;
- The footer CTA was repaired to look better. Thanks for checking it;
- I am using a `cover` instead of an _image_ to ensure the aspect ratio is set to 3:2. We can't control AR on images; we can only upload cut ones. But that would be a hack, and users wouldn't follow it. Do you think this is a problem?
- Made the author's text into a Heading;
- Added a main tag to all the `Content wrappers`;
- Removed the unused patterns: single and page.php;
- Changed the Required PHP to 7.2;
- Added the credit to the image that was missing;
- Added the extra tags you mentioned.

iamtakashi commented:
@henriqueiamarino, thanks for the update! Most of the issues were addressed correctly 👍 

* Thanks for removing the custom duotones, but it's still used in [here](https://github.com/Automattic/themes/pull/8637/files#diff-ba990d44b546e7c8c6b5f423e0ed9ad902fbcc982f7fbecba6e763fb5379e91cR11) and [here](https://github.com/Automattic/themes/pull/8637/files#diff-04c325b7a9520bf50e1dfbcd6dffd29998a4568d74ad24f209139d0bee6e54aaR100).
* The credit for the shadow image is missing.
* The height of the button is now different between the block variations. You have fixed the same issue [here](https://github.com/Automattic/themes/issues/8500#issuecomment-2555663994). However, this time, the border thickness is 2px. 
![CleanShot 2025-01-30 at 12 18 31@2x](https://github.com/user-attachments/assets/97599bb2-144e-44fd-b15b-a60df14a388b)

henriqueiamarino committed:
Lexionic: removes duotones from variations

iamtakashi committed:
Made the outline version of button block same height as default
